bokehjs/src/lib/models/transforms/jitter.ts

```diff
@@ -35,10 +36,28 @@ export class Jitter extends RangeTransform {
   }
 
   override v_compute(xs0: Arrayable<number | Factor>): Arrayable<number> {
-    if (this.previous_values != null && this.previous_values.length == xs0.length)
-      return this.previous_values
-    this.previous_values = super.v_compute(xs0)
-    return this.previous_values
+    let xs: Arrayable<number>
+    if (super.range instanceof FactorRange)
+      xs = super.range.v_synthetic(xs0)
+    else if (isArrayableOf(xs0, isNumber))
+      xs = xs0
+    else
+      throw new Error("unexpected")
+
+    if (this.previous_offsets != null && this.previous_offsets.length == xs.length) {
```

Comment:
This could be simplified to `if (this.previous_offsets?.length == xs.length)`. Also, `previous_offsets` is implicitly undefined, but we would prefer this to be explicit, so change the type to `previous_offsets: Arrayable<number> | undefined`.

Reply:
Changed